src/main/services/pre-api/pre-client.ts

```diff
@@ -25,7 +31,55 @@ export class PreClient {
 
       return response.data as Recording;
     } catch (e) {
-      return null;
+      if (e.response.status === 404) {
```

Comment:
```suggestion
      if (e.response?.status === 404) {
```
I think the response may not always get set (maybe not an error with axios but with marshalling `response.data` as a `Recording` for example)
the `?` will allow that to work if present or not flip out if it's not.